### ISSUE TITLE
Confusing getting started experience [docs]

### ISSUE THREAD

**Issue by pseudo-su** [redhat-developer/vscode-java#435]
### Confusing getting started experience [docs]

As someone who doesn't know much about java I needed to setup a dev environment for an existing Java project.

I'm used to writing & debugging front-end/nodejs/rust/golang in vscode ([Hooray LSP 😍 🎉](https://github.com/Microsoft/language-server-protocol))

It took me a while to figure out why this extension wasn't working initially and I thought I'd put the process I went through here with the hope that it can be used to make the readme/docs clearer for someone who doesn't know much about the Java ecosystem.

#### Environment

* **Operating System**: MacOS High Sierra
* **JDK version**: adopt@1.8.162-00 (Installed using [jabba](https://github.com/shyiko/jabba))
* **Visual Studio Code version**: 1.20.0
* **Java extension version**: 0.18.1

#### TL;DR

1. This plugin was generating a bunch of normal eclipse project files & structure. I didn't realise that because the plugin uses the eclipse Language Server implementation it also acts like and relies upon eclipse-y things.
1. While this plugin does create a lot of eclipse-specific files it doesn't create `.classpath` which is needed for the the eclipse tooling/language-server to work. Google told me the error I was getting was because it couldn't find the JDK so I spent a fair bit of time fiddling with my `JAVA_HOME` in vain attempting to make something work. In the end I needed to manually add `apply plugin: 'eclipse'` to `build.gradle`

##### Things I needed to know

Understand that this plugin acts like eclipse in a lot of ways so solutions related to eclipse are likely to be relevant for this extension as well.

1. add `apply plugin: 'eclipse'` to `build.gradle` (or equivalent for maven).
2. Initialize eclipse :
    ```bash
    ./gradlew cleanEclipse eclipse
    # OR
    mvn eclipse:clean eclipse:eclipse
    ```

#### Sources of confusion getting this extension working

* Installed the extensions from the vscode marketplace
  > code --install-extensions=vscjava.vscode-java-pack
* I was a bit confused as to why this extension was creating files and folders in my project directory. Didn't know where to look to find out why/where they came from (didn't realise this extension imitates eclipse).
  * `/.project`: [Project Description File](https://help.eclipse.org/oxygen/index.jsp?topic=/org.eclipse.platform.doc.isv/reference/misc/project_description_file.html)
  * `/.settings` (can't find any docs link)
  * `/bin` (can't find any docs link)
* Ignoring all the generated files for the moment I tried to figure out instead why the I was getting [this `java.lang.Object cannot be resolved` error](https://stackoverflow.com/questions/18075343/java-project-in-eclipse-the-type-java-lang-object-cannot-be-resolved-it-is-ind) at the top of my main file.
* Thought it wasn't working because it couldn't detect my JDK / JAVA_HOME, presumed it was something to do with the extension configuration but the only seemingly useful settings entry is `java.home` and it didn't help.
* Thought it might be something to do with the fact the project uses gradle because the docs mention it's not fully supported.
* Because I kept running into eclipse-related problems I thought I might have to install the eclipse IDE to get this working (which I was pretty sad about because the whole point was to avoid installing a Java IDE 😄). I tried to install Eclipse but it gave me an error about needing to install a legacy 1.6 JDK/JRE which I didn't want to do.
* Just as I was about to give up and wait for an IntelliJ license key (😢🤕) I found the eclipse gradle plugin and without really knowing what it would do, installed it. Realised it added `cleanEclipse` & `eclipse` tasks, which added a `.classpath` file and suddenly everything started working (errors, warnings, debugging etc) 🎉🎉🎉🎉.
* Realised that `Eclipse IDE` does not necessarily equal `eclipse` and this extension has been eclipse masquerading as vscode this whole time (kind of).

#### Additional Information

Now that I know what I actually need to do to get this extension working it's **super awesome**! I'm very excited about potentially not having to install IntelliJ/eclipseIDE/netbeans etc 😁.

**Comment by gorkem** [comment]
Thank you for this information it is very helpful. 

Normally, you should not need the `apply plugin: 'eclipse' to build.gradle` for the extension to work. It does have built-in gradle support and should have recognized the buildpaths. Do you have an example project that you can share so that we can try to understand what went wrong.

You definitely do not need an eclipse to run vscode-java, we do have a small eclipse subset that bundles the Java support from eclipse. Hence the `.classpath` and friends. Where do you think the information for these temporary files would be helpful?

**Comment by pseudo-su** [comment]
The project/repository I had the issue with is private so I can't share it directly. I thought I'd try and reconstruct the problem but after going back to the initial repo and building from scratch I no longer have any issue.

It now functions exactly as you say it should, generates a `.classpath` with no need for the separate `cleanEclipse` and `eclipse` gradle tasks.

When I was having issues I was looking at the `README.md` that shows in vscode when you go to the extensions tab. It seems like there's a lot more info there now than what I remember (or maybe i'm just going crazy).

Either way my initial problem looks to be solved and the documentation seems to explain things well.

**Comment by erihanse** [comment]
I'm still a little confused regarding how to set up a simple hello world from scratch. I'm used to compiling and running using a cli, and eclipse. Maven seems a little overkill. 

What I've been doing is opening the eclipse IDE, creating a new project, and continuing from there in vs code. I'm curious how I can do the same thing without using eclipse IDE? In case I'm asking the wrong questions here, how can I get something minimalistic like this to run in vs code?
![image](https://user-images.githubusercontent.com/9251274/38883289-baca141e-426c-11e8-9b90-291cd21d29e9.png)
